src/components/utils/popover.ts

```diff
@@ -381,30 +480,92 @@ export default class Popover extends EventsDispatcher<PopoverEvent> {
       }));
     }
 
+    if (item.isActive) {
+      el.classList.add(Popover.CSS.itemActive);
+    }
+
     return el;
   }
 
   /**
    * Item click handler
    *
    * @param itemEl - clicked item
+   * @param event - click event
    */
-  private itemClicked(itemEl: HTMLElement): void {
+  private itemClicked(itemEl: HTMLElement, event: MouseEvent): void {
     const allItems = this.nodes.wrapper.querySelectorAll(`.${Popover.CSS.item}`);
     const itemIndex = Array.from(allItems).indexOf(itemEl);
     const clickedItem = this.items[itemIndex];
 
-    clickedItem.onClick(clickedItem);
+    this.cleanUpConfirmationState();
+
+    if (clickedItem.confirmation) {
+      /** Save root item requiring confirmation to restore original state on popover hide */
+      if (this.itemRequiringConfirmation === null) {
+        this.itemRequiringConfirmation = clickedItem;
+      }
+      this.items[itemIndex] = clickedItem.confirmation;
```

Comment:
what if some data was not passed? For example, we override only the `label`, but icon should be preserved. 

We need to merge the confirmatuib config with the root?

Reply:
added merging configs